test/jdk/com/sun/net/httpserver/HttpsParametersClientAuthTest.java

```diff
@@ -0,0 +1,94 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.net.InetSocketAddress;
+
+import javax.net.ssl.SSLParameters;
+
+import com.sun.net.httpserver.HttpsConfigurator;
+import com.sun.net.httpserver.HttpsParameters;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/*
+ * @test
+ * @bug 8326381
+ * @summary verifies that the setNeedClientAuth() and setWantClientAuth()
+ *          methods on HttpsParameters class work as expected
+ * @run junit HttpsParametersClientAuthTest
+ */
+public class HttpsParametersClientAuthTest {
+
+    /**
+     * verifies that {@link HttpsParameters#setNeedClientAuth(boolean)}
+     * and {@link HttpsParameters#setWantClientAuth(boolean)} methods work
+     * as expected
+     */
+    @Test
+    public void testClientAuth() throws Exception {
+        // test default values
+        HttpsParameters defaultParams = new Params();
+        assertFalse(defaultParams.getNeedClientAuth(),
+                "needClientAuth was expected to be false but wasn't");
+        assertFalse(defaultParams.getWantClientAuth(),
+                "wantClientAuth was expected to be false but wasn't");
+
+        // needClientAuth = true and thus wantClientAuth = false
+        HttpsParameters needClientAuthParams = new Params();
+        needClientAuthParams.setNeedClientAuth(true);
+        assertTrue(needClientAuthParams.getNeedClientAuth(),
+                "needClientAuth was expected to be true but wasn't");
+        assertFalse(needClientAuthParams.getWantClientAuth(),
+                "wantClientAuth was expected to be false but wasn't");
+
+        // wantClientAuth = true and thus needClientAuth = false
+        HttpsParameters wantClientAuthParams = new Params();
+        wantClientAuthParams.setWantClientAuth(true);
+        assertTrue(wantClientAuthParams.getWantClientAuth(),
+                "wantClientAuth was expected to be true but wasn't");
+        assertFalse(wantClientAuthParams.getNeedClientAuth(),
+                "needClientAuth was expected to be false but wasn't");
+    }
```

Comment:
Is it necessary to check the states on the following cases?

```
HttpsParameters params = new Params();
params.setNeedClientAuth(true);
params.setWantClientAuth(false);
```

```
HttpsParameters params = new Params();
params.setWantClientAuth(true);
params.setNeedClientAuth(false);
```

Reply:
I don't have preference, especially since we now decided to deprecate these methods. But if you prefer we add those, then let me know and I can update the test tomorrow.

Reply:
I thought this fix made different states of `wantClientAuth` and `needClientAuth` for those cases.
So, it may be better to check if the states in those cases are expected, though they should be corner cases.